kratos/geometries/geometry.h

```diff
@@ -2634,31 +2634,15 @@ class Geometry
     *          1 -> inside
     *          2 -> on the boundary
     */
+    KRATOS_DEPRECATED_MESSAGE("This method is deprecated. Use either \'ClosestPointLocalToLocalSpace\' or \'ClosestPointGlobalToLocalSpace\' instead. Please note that \'rClosestPointGlobalCoordinates\' returns unmodified original value.")
     virtual int ClosestPoint(
         const CoordinatesArrayType& rPointGlobalCoordinates,
         CoordinatesArrayType& rClosestPointGlobalCoordinates,
         CoordinatesArrayType& rClosestPointLocalCoordinates,
         const double Tolerance = std::numeric_limits<double>::epsilon()
     ) const
     {
-        // 1. Make projection on geometry
-        const int projection_result = ProjectionPointGlobalToLocalSpace(
-            rPointGlobalCoordinates,
-            rClosestPointLocalCoordinates,
-            Tolerance);
-
-        if (projection_result == 1) {
-            // 2. If projection converged check if solution lays
-            // within the boundaries of this geometry
-            // Returns either 0, 1 or 2
-            // Or -1 if IsInsideLocalSpace failed
-            return IsInsideLocalSpace(
-                rClosestPointLocalCoordinates,
-                Tolerance);
-        } else {
-            // Projection failed
-            return -1;
-        }
+        return ClosestPointGlobalToLocalSpace(rPointGlobalCoordinates, rClosestPointLocalCoordinates, Tolerance);
```

Comment:
```suggestion
        const int result = ClosestPointGlobalToLocalSpace(rPointGlobalCoordinates, rClosestPointLocalCoordinates, Tolerance);
        
        if (result == 1) {
            this->GlobalCoordinates(rClosestPointGlobalCoordinates, local_coordinates);
        }

        return result;
```

Not that this is of major importance as it will be removed. But I think the correct term is as this.

Reply:
Agree. But note that I'm keeping the previous behavior, which was wrong (see the  deprecation message).

Reply:
ok, no worries, then.